lab_3_generate_by_ngrams/main.py

```diff
@@ -270,7 +391,25 @@ def run(self, seq_len: int, prompt: str) -> Optional[str]:
         In case of corrupt input arguments or methods used return None,
         None is returned
         """
-
+        if not(isinstance(seq_len, int)
+               and isinstance(prompt, str) and prompt):
+            return None
+        encoded_prompt = self._text_processor.encode(prompt)
+        len_of_context = self._model.get_n_gram_size()
+        if not (encoded_prompt and len_of_context):
+            return None
+        continued_sequence = prompt
+        for i in range(seq_len):
+            next_tokens = self._model.generate_next_token(encoded_prompt[:-len_of_context + 1])
+            if not next_tokens:
+                break
+            max_freq = max(next_tokens.values())
+            next_token = next_tokens[max_freq]
+            continued_sequence += str(self._text_processor.decode(next_token))
```

Comment:
No need to decode at every step, you can generate the full encoded sequence and then decode and post-process it right before the return

Reply:
Fixed, but this function is still not working